picard/webservice/__init__.py

```diff
@@ -260,6 +260,64 @@ def _init_headers(self):
         super()._init_headers()
 
 
+class RequestTask(namedtuple('RequestTask', 'hostkey func priority')):
+
+    @staticmethod
+    def from_request(request, func):
+        # priority is a boolean
+        return RequestTask(request.get_host_key(), func, int(request.priority))
+
+
+class RequestPriorityQueue:
+
+    def __init__(self, ratecontrol):
+        self._queues = defaultdict(lambda: defaultdict(deque))
+        self._ratecontrol = ratecontrol
+        self._count = 0
+
+    def count(self):
+        return self._count
+
+    def add_task(self, task, important=False):
+        (hostkey, func, prio) = task
+        queue = self._queues[prio][hostkey]
+        if important:
+            queue.appendleft(func)
+        else:
+            queue.append(func)
+        self._count += 1
+        return RequestTask(hostkey, func, prio)
+
+    def remove_task(self, task):
+        hostkey, func, prio = task
+        try:
+            self._queues[prio][hostkey].remove(func)
+            self._count -= 1
+        except Exception as e:
+            log.debug(e)
+
+    def run_ready_tasks(self):
+        delay = sys.maxsize
+        for prio in sorted(self._queues.keys(), reverse=True):
+            prio_queue = self._queues[prio]
+            if not prio_queue:
+                del(self._queues[prio])
```

Comment:
I just noticed this syntax but according to https://docs.python.org/3/reference/simple_stmts.html#the-del-statement `del x` is the correct syntax, `del(x)` is interpreted as `del (x)`, and parentheses are ignored (see https://docs.python.org/3/reference/simple_stmts.html#grammar-token-python-grammar-target_list).
Elsewhere in the code we use `del x`, is there a specific reason to not do the same here?